### PR TITLE
Re-added bootstrap.sh, updated README

### PR DESCRIPTION
**Pull request by ilijapuaca**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,30 +95,43 @@ Our hope is to see this project quickly grow into a living part of the core deck
...
 ### gcc
 
-For Mac OSX:
+For macOS:
 
 ```sh
 brew install gcc cmake clang-format lcov
 ```
 
+## Quickstart
```
Comment (ibgreen) [inline README.md:124]: `Quickstart` section towards the top... first subheading just under `Setting up a development environment`?

File: README.md [hunk 2]
```diff
@@ -95,30 +95,43 @@ Our hope is to see this project quickly grow into a living part of the core deck
...
+
+- `scripts/bootstrap.sh` to fetch the dependencies and perform setup
+- `scripts/build-clang` or `scripts/build-gcc` to build
+
 ## Dependencies
 
+Development can be done on macOS which is the primary environment, or Linux, which is supported mainly for CI testing. (Windows is not a supported dev env.)
```
Comment (ibgreen) [inline README.md:131]: This sentence should probably stay as the first sentence in setting up a development environment?

File: README.md [hunk 3]
```diff
@@ -95,30 +95,43 @@ Our hope is to see this project quickly grow into a living part of the core deck
 
 ## Setting up a Development Environment
 
-Development can be done on macOS which is the primary environment, or Linux, which is supported mainly for CI testing. (Windows is not a supported dev env.)
```
Comment (ibgreen) [inline README.md:98]: why not keep this here?

File: README.md [hunk 4]
```diff
@@ -95,30 +95,43 @@ Our hope is to see this project quickly grow into a living part of the core deck
...
+## Quickstart
+
+- `scripts/bootstrap.sh` to fetch the dependencies and perform setup
+- `scripts/build-clang` or `scripts/build-gcc` to build
+
 ## Dependencies
 
+Development can be done on macOS which is the primary environment, or Linux, which is supported mainly for CI testing. (Windows is not a supported dev env.)
+
+All the dependencies for macOS and Linux are bundled in a dependency repository that's being used as a submodule. Running `scripts/bootstrap.sh` fetches all the submodules.
```
Comment (ibgreen) [inline README.md:133]: In my mind the `bootstrap` script is not restricted to initializing the submodule. It does any setup that needs to be done after pulling master (it just so happens that right now this is the only thing it does). 

So I would introduce it higher up .

Add: "alternatively, call `git submodule update --init --recursive`". Do we even need that `recursive` flag or just good habit?

File: README.md [hunk 5]
```diff
@@ -95,30 +95,43 @@ Our hope is to see this project quickly grow into a living part of the core deck
...
-For Mac OSX:
+For macOS:
 
 ```sh
 brew install gcc cmake clang-format lcov
 ```
 
+## Quickstart
+
+- `scripts/bootstrap.sh` to fetch the dependencies and perform setup
```
Comment (ibgreen) [inline README.md:126]: Maybe add a note that the intention is that bootstrap can be called after pulling a branch to ensure env is updated.

File: README.md (hunk 4 above)
Comment (ilijapuaca) [reply README.md:133]: Right now we don't, but it's a good habit in case we ever have nested submodules

**Pull request merged**